pkg/datagatherer/versionchecker/versionchecker.go

```diff
@@ -257,6 +277,26 @@ type containerResult struct {
 
 // Fetch retrieves cluster information from GKE.
 func (g *DataGatherer) Fetch() (interface{}, error) {
+	// Get nodes information to update version-checker architecture structure
+	rawNodes, err := g.nodeDynamicDg.Fetch()
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch nodes, %s", err)
+	}
+	nodes, ok := rawNodes.(*unstructured.UnstructuredList)
+	if !ok {
+		return nil, fmt.Errorf("failed to parse nodes loaded from DataGatherer")
+	}
+	for _, v := range nodes.Items {
+		var node v1.Node
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(v.Object, &node)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse node from unstructured data")
+		}
+		if err = g.nodeArchitecture.Add(&node); err != nil {
```

Comment:
I might add a comment here explaining the impact this has on version checker. My guess is that this gives VC a means to lookup the arch for the node of a given pod to determine the correct latest version?